src/main/java/org/jabref/gui/libraryproperties/LibraryPropertiesDialogView.java

```diff
@@ -0,0 +1,214 @@
+    @FXML private ComboBox<Charset> encoding;
+    @FXML private TextField generalFileDirectory;
+    @FXML private Button browseGeneralFileDir;
+    @FXML private TextField userSpecificFileDirectory;
+    @FXML private Button browseUserSpefiicFileDir;
+    @FXML private VBox saveOrdervbox;
+    @FXML private ToggleGroup saveOrderToggleGroup;
+    @FXML private RadioButton saveInOriginalOrder;
+    @FXML private RadioButton saveInSpecifiedOrder;
+    @FXML private CheckBox protect;
+    private SaveOrderConfigDisplay saveOrderPanel;
+    private BasePanel panel;
+    private LibraryPropertiesDialogViewModel viewModel;
+    private final DialogService dialogService;
+    @Inject private PreferencesService preferencesService;
+    private FieldFormatterCleanupsPanel fieldFormatterCleanupsPanel;
+    private Object oldSaveOrderConfig;
+
+    public LibraryPropertiesDialogView(BasePanel panel, DialogService dialogService) {
+        this.dialogService = dialogService;
+        this.panel = panel;
+        ViewLoader.view(this)
+                  .load()
+                  .setAsDialogPane(this);
+
+        setResultConverter(btn -> {
+            if (btn == ButtonType.OK) {
+                storeSettings();
+            }
+            return null;
+        });
+
+        setTitle(Localization.lang("Library properties"));
+
+    }
+
+    @FXML
+    private void initialize() {
+        viewModel = new LibraryPropertiesDialogViewModel(panel, dialogService, preferencesService);
+
+        generalFileDirectory.textProperty().bindBidirectional(viewModel.generalFileDirectoryPropertyProperty());
+        userSpecificFileDirectory.textProperty().bindBidirectional(viewModel.userSpecificFileDirectoryProperty());
+
+        encoding.itemsProperty().bind(viewModel.encodingsProperty());
+        encoding.valueProperty().bindBidirectional(viewModel.selectedEncodingProperty());
+        saveOrderPanel = new SaveOrderConfigDisplay();
+        saveOrdervbox.getChildren().add(saveOrderPanel.getJFXPanel());
+        fieldFormatterCleanupsPanel = new FieldFormatterCleanupsPanel(Localization.lang("Enable save actions"),
+                                                                      Cleanups.DEFAULT_SAVE_ACTIONS);
+
+        contentVbox.getChildren().add(fieldFormatterCleanupsPanel);
+        saveInOriginalOrder.selectedProperty().bindBidirectional(viewModel.saveInOriginalProperty());
+        saveInSpecifiedOrder.selectedProperty().bindBidirectional(viewModel.saveInSpecifiedOrderProperty());
+
+        protect.selectedProperty().bindBidirectional(viewModel.libraryProtectedProperty());
+        saveOrderPanel.getJFXPanel().disableProperty().bind(viewModel.saveInOriginalProperty());
+
+        setValues();
+        updateEnableStatus();
+    }
+
+    private void setValues() {
+
+        viewModel.setValues();
+        Optional<SaveOrderConfig> storedSaveOrderConfig = panel.getBibDatabaseContext().getMetaData().getSaveOrderConfig();
```

Comment:
This logic should be moved to the viewmodel. For this you might need to work a bit on the save order and cleanup panel so that these use javafx properties to manage their values. But this is needed anyway in the long run...